app/lib/features/super_invites/pages/create_super_invite_page.dart

```diff
@@ -0,0 +1,375 @@
+import 'package:logging/logging.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+
+final _log = Logger('a3::super_invites::create');
+
+class CreateSuperInvitePage extends ConsumerStatefulWidget {
+  static Key tokenFieldKey = const Key('super-invites-create-token-token');
+  static Key createDmKey = const Key('super-invites-create-token-create-dm');
+  static Key addSpaceKey = const Key('super-invites-create-token-add-space');
+  static Key addSubmenu = const Key('super-invites-create-token-add-submenu');
+  static Key addChatKey = const Key('super-invites-create-token-add-chat');
+  static Key submitBtn = const Key('super-invites-create-submitBtn');
+  static Key deleteBtn = const Key('super-invites-create-delete');
+  static Key deleteConfirm = const Key('super-invites-create-delete-confirm');
+
+  final SuperInviteToken? token;
+
+  const CreateSuperInvitePage({
+    super.key,
+    this.token,
+  });
+
+  @override
+  ConsumerState<CreateSuperInvitePage> createState() =>
+      _CreateSuperInvitePageState();
+}
+
+class _CreateSuperInvitePageState extends ConsumerState<CreateSuperInvitePage>
+    with SingleTickerProviderStateMixin {
+  final TextEditingController _tokenController = TextEditingController();
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  late SuperInvitesTokenUpdateBuilder tokenUpdater;
+  bool isEdit = false;
+  bool _initialDmCheck = false;
+  List<String> _roomIds = [];
+
+  @override
+  void initState() {
+    super.initState();
+    _initializeToken();
+  }
+
+  Future<void> _initializeToken() async {
+    final token = widget.token;
+    if (token != null) {
+      // given an update builder we are in an edit mode
+      isEdit = true;
+      _tokenController.text = token.token();
+      _roomIds = asDartStringList(token.rooms());
+      _initialDmCheck = token.createDm();
+      tokenUpdater = token.updateBuilder();
+    } else {
+      final superInvites = await ref.read(superInvitesProvider.future);
+      tokenUpdater = superInvites.newTokenUpdater();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final lang = L10n.of(context);
+    return Scaffold(
+      appBar: _buildAppBarView(lang),
+      body: _buildBodyView(lang),
+    );
+  }
+
+  AppBar _buildAppBarView(L10n lang) {
+    return AppBar(
+      title: Text(
+        isEdit ? lang.editInviteCode : lang.createInviteCode,
+      ),
+    );
+  }
+
+  Widget _buildBodyView(L10n lang) {
+    return SingleChildScrollView(
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 16),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: [
+            isEdit && widget.token != null
+                ? InviteListItem(
+                    inviteToken: widget.token!,
+                    cardMargin:
+                        EdgeInsets.symmetric(horizontal: 0, vertical: 12),
+                  )
+                : Form(
```

Comment:
May I suggest to split this into a more readable format, which also removes the `!` (which we want to avoid whenever possible)

```suggestion
    final token = widget.token;
    return SingleChildScrollView(
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            if (isEdit && token != null)
               InviteListItem(
                    inviteToken: token,
                    cardMargin:
                        EdgeInsets.symmetric(horizontal: 0, vertical: 12),
                  )
            else
                Form(
```